### PR TITLE
AYS-192 | User Detail Flow Has Been Created 

### PR DESCRIPTION
**Pull request by egehanasal**
### Checklist

Before submitting your pull request, ensure the following:

- [X] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [X] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [X] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [X] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [X] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [X] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [X] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

**Review by AbdullahCelikcode — DISMISSED** [review]

File: src/main/java/org/ays/auth/util/exception/AysUserNotExistException.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.ays.auth.util.exception;
+
+import org.ays.common.util.exception.AysNotExistException;
+
+import java.io.Serial;
+
+/**
+ * Exception to be thrown when a user with a given id does not exist.
+ */
+public final class AysUserNotExistException extends AysNotExistException {
```
Comment (agitrubard) [inline src/main/java/org/ays/auth/util/exception/AysUserNotExistException.java:10]:
```suggestion
public final class AysUserNotExistByIdException extends AysNotExistException {
```

File: src/test/java/org/ays/auth/controller/AysUserEndToEndTest.java [hunk 2]
```diff
@@ -154,4 +164,60 @@ void givenValidUserListRequest_whenUsersFoundForAdmin_thenReturnAysPageResponseO
...
+                        .value(user.getId()))
+                .andExpect(AysMockResultMatchersBuilders.response("firstName")
+                        .value(user.getFirstName()))
+                .andExpect(AysMockResultMatchersBuilders.response("status")
+                        .isNotEmpty())
+                .andExpect(AysMockResultMatchersBuilders.response("roles[*].id")
+                        .isNotEmpty())
+                .andExpect(AysMockResultMatchersBuilders.response("roles[*].name")
+                        .isNotEmpty());
+    }
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/controller/AysUserEndToEndTest.java:221]: Burada bütün fieldların varlığını kontrol edebilir miyiz?

File: src/test/java/org/ays/auth/service/impl/AysUserReadServiceImplTest.java [hunk 3]
```diff
@@ -105,4 +108,50 @@ void givenValidListRequest_whenUsersNotFound_thenReturnAysPageOfUsers() {
...
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(mockUser, result);
+
+        // Verify
+        Mockito.verify(userReadPort, Mockito.times(1))
+                .findById(Mockito.anyString());
+    }
+
+    @Test
+    void givenValidUserId_whenUserNotFoundById_thenThrowAysUserNotExistException() {
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/service/impl/AysUserReadServiceImplTest.java:137]:
```suggestion
    void givenValidId_whenUserNotFoundById_thenThrowAysUserNotExistByIdException() {
```

File: src/test/java/org/ays/auth/service/impl/AysUserReadServiceImplTest.java [hunk 4]
```diff
@@ -105,4 +108,50 @@ void givenValidListRequest_whenUsersNotFound_thenReturnAysPageOfUsers() {
                 .getInstitutionId();
     }
 
+
+    @Test
+    void givenValidUserId_whenUserFoundById_thenReturnUser() {
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/service/impl/AysUserReadServiceImplTest.java:113]:
```suggestion
    void givenValidId_whenUserFoundById_thenReturnUser() {
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/org/ays/auth/util/exception/AysUserNotExistException.java (hunk 1 above)
Comment (egehanasal) [reply src/main/java/org/ays/auth/util/exception/AysUserNotExistException.java]: değiştirdim

File: src/test/java/org/ays/auth/controller/AysUserEndToEndTest.java (hunk 2 above)
Comment (egehanasal) [reply src/test/java/org/ays/auth/controller/AysUserEndToEndTest.java:231]: güncelledim

File: src/test/java/org/ays/auth/service/impl/AysUserReadServiceImplTest.java (hunk 3 above)
Comment (egehanasal) [reply src/test/java/org/ays/auth/service/impl/AysUserReadServiceImplTest.java]: değiştirdim

**Review by agitrubard — APPROVED** [review]

**Review by MenekseYuncu — APPROVED** [review]

**Pull request merged**